source/source_test.go

```diff
@@ -0,0 +1,190 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/config"
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/models"
+	"github.com/conduitio-labs/conduit-connector-gcp-pubsub/source/mock"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/golang/mock/gomock"
+	"github.com/matryer/is"
+)
+
+func TestSource_ConfigureSuccess(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	s := Source{}
+
+	err := s.Configure(context.Background(), map[string]string{
+		models.ConfigPrivateKey:     "-----BEGIN PRIVATE KEY-----\nMII\n-----END PRIVATE KEY-----\n",
+		models.ConfigClientEmail:    "test@pubsub-test.iam.gserviceaccount.com",
+		models.ConfigProjectID:      "pubsub-test",
+		models.ConfigSubscriptionID: "conduit-subscription-b595b388-7a97-4837-a180-380640d9c43f",
+	})
+	is.NoErr(err)
+	is.Equal(s.cfg, config.Source{
+		General: config.General{
+			PrivateKey:  "-----BEGIN PRIVATE KEY-----\nMII\n-----END PRIVATE KEY-----\n",
+			ClientEmail: "test@pubsub-test.iam.gserviceaccount.com",
+			ProjectID:   "pubsub-test",
+		},
+		SubscriptionID: "conduit-subscription-b595b388-7a97-4837-a180-380640d9c43f",
+	})
+}
+
+func TestSource_ConfigureFail(t *testing.T) {
```

Comment:
Got it.
I've renamed it to `TestSource_ConfigureFailPrivateKeyIsRequired`.
But, TBH, I've just wanted to show the fail case, because all of these cases were added in [general_test.go:65](https://github.com/conduitio-labs/conduit-connector-gcp-pubsub/blob/develop/config/general_test.go#L65).

Reply:
Oh, I see your point now. You wanted to make sure that Source.Configure() fails, if configuration is invalid (and doesn't ignore it or something). In that case, I agree with you.